src/commands/force/source/deploy.ts

```diff
@@ -225,37 +225,79 @@ export class Deploy extends DeployCommand {
+      }
+      this.ux.log('');
+      const successHeader: string = chalk.green('Result');
+      const failureHeader: string = chalk.red('Result');
+      const header: string = data.success ? successHeader : failureHeader;
+      this.ux.styledHeader(header);
+      this.ux.log('');
+    }
+
+    const successfulComponentsMessage: string = data.checkOnly
+      ? `Components checked:  ${data.numberComponentsDeployed}`
+      : `Components deployed:  ${data.numberComponentsDeployed}`;
+    this.ux.log('');
+    this.ux.log(`Status:  ${data.status}`);
+    this.ux.log(`jobid:  ${data.id}`);
+    this.ux.log(`Component errors:  ${data.numberComponentErrors}`);
+    this.ux.log(successfulComponentsMessage);
+    this.ux.log(`Components total:  ${data.numberComponentsTotal}`);
+    this.ux.log(`Tests errors:  ${data.numberTestErrors}`);
+    this.ux.log(`Tests completed:  ${data.numberTestsCompleted}`);
+    this.ux.log(`Tests total:  ${data.numberTestsTotal}`);
+    this.ux.log(`Check only: ${data.checkOnly}`);
+    this.ux.log('');
+  }
+
+  private progress(deploy: MetadataApiDeploy, progressBar = true): void {
     const startProgressBar = once((componentTotal: number) => {
       this.progressBar.start(componentTotal);
     });
-
     deploy.onUpdate((data) => {
-      // the numCompTot. isn't computed right away, wait to start until we know how many we have
-      if (data.numberComponentsTotal) {
-        startProgressBar(data.numberComponentsTotal + data.numberTestsTotal);
-        this.progressBar.update(data.numberComponentsDeployed + data.numberTestsCompleted);
-      }
-
-      // the numTestsTot. isn't computed until validated as tests by the server, update the PB once we know
-      if (data.numberTestsTotal && data.numberComponentsTotal) {
-        this.progressBar.setTotal(data.numberComponentsTotal + data.numberTestsTotal);
+      if (!progressBar) {
+        // If not progress bar. Print the status in normal formatt
+        this.printStatus(data);
+      } else {
+        // the numCompTot. isn't computed right away, wait to start until we know how many we have
+        if (data.numberComponentsTotal) {
+          startProgressBar(data.numberComponentsTotal + data.numberTestsTotal);
+          this.progressBar.update(data.numberComponentsDeployed + data.numberTestsCompleted);
+        }
+        // the numTestsTot. isn't computed until validated as tests by the server, update the PB once we know
+        if (data.numberTestsTotal && data.numberComponentsTotal) {
+          this.progressBar.setTotal(data.numberComponentsTotal + data.numberTestsTotal);
+        }
       }
     });
-
     // any thing else should stop the progress bar
     deploy.onFinish((data) => {
-      // the final tick of `onUpdate` is actually fired with `onFinish`
-      this.progressBar.update(data.response.numberComponentsDeployed + data.response.numberTestsCompleted);
-      this.progressBar.stop();
+      if (progressBar) {
```

Comment:
can we change this so it's the same conditional statement between both listeners. e.g. always use `if(progressBar)`